tensorflow/core/common_runtime/bfc_allocator.h

```diff
@@ -246,12 +246,12 @@ class BFCAllocator : public Allocator {
       std::swap(handles_, other.handles_);
     }
 
-    int IndexFor(const void* p) const {
+    size_t IndexFor(const void* p) const {
       std::uintptr_t p_int = reinterpret_cast<std::uintptr_t>(p);
       std::uintptr_t base_int = reinterpret_cast<std::uintptr_t>(ptr_);
       DCHECK_GE(p_int, base_int);
       DCHECK_LT(p_int, base_int + memory_size_);
-      return static_cast<int>(((p_int - base_int) >> kMinAllocationBits));
+      return static_cast<size_t>(((p_int - base_int) >> kMinAllocationBits));
```

Comment:
I don't see this can overflow: to make the result larger than `2^31-1`, `(p_int - base_int)` has to be at least `2^(31+8)=512GB`. Or do you have an example to reproduce the error?

Reply:
That's right. But since base_int is the starting address of the memory pool and p_int could be anywhere, the offset of base_int and p_int could overflow when the size of the memory pool exceeds 512 GB. The overflow happens when we conduct large-image (10000x10000) DNN training that requires much amount of memory.

Reply:
Here is the example to reproduce the error:
https://gist.github.com/chenchc/71cc4772d45873ee71b04d13276da3ba

Reply:
I see, so you're using unified memory, that makes sense. Thanks for the repo.

Reply:
Yeah, that's the point. Thanks for the review.